packages/qwik/src/core/GLOSSARY.md

```diff
@@ -62,3 +62,90 @@ Here is the basic flow of data: JSX -> vDOM -> `vnode_diff` -> VNode -> DOM
 - `vnode_diff`: The algorithm that compares two vDOMs and produces a list of changes.
 - VNode: A layer on top of DOM which can be deserialized from SSR, and which understands Virtual Nodes and properties on those nodes.
 - DOM: The actual browser DOM.
+
+vNodes are Qwik's translation layer between your JSX and the actual HTML. They act like sticky notes, helping Qwik track and manage changes efficiently.
+
+```tsx
+<Component>
+  <button />
+</Component>
+```
+
+`Component` and `Signal` are examples of these sticky notes. Signal vNodes are particularly interesting as they create boundaries between normal text nodes, marking regions that can change.
+
+When Qwik updates your UI, it follows these steps:
+
+1. First, it looks at your sticky notes (vNodes) to see the current state
+2. Then, it compares this with your desired changes (JSX)
+3. Finally, it records the necessary changes in a journal, which will be applied to the HTML in a separate step
+
+## How the Diff Works
+
+The diff process uses two key pieces:
+
+- `jsxNode`: What you want your component to look like (input)
+- `vStartNode`: A snapshot of what your component currently looks like (output)
+
+> The diff only looks at the specific component or signal being updated, not the entire page.
```

Comment:
it depends on what was changed actually. For CSR render we look at the entire page :D